permiso/src/main/java/com/greysonparrelli/permiso/PermisoDialogFragment.java

```diff
@@ -48,16 +55,102 @@ public static PermisoDialogFragment newInstance(@Nullable String title, @NonNull
         return dialogFragment;
     }
 
+    public static class Builder {
+        private int titleId = 0;
+        private String title = null;
+        private int buttonTextId = 0;
+        private String message = null;
+        private int messageId = 0;
+        // by default, interpret the dialog msg body as string text
+        private boolean interpretHtml = false;
+        private String buttonText = null;
+
+
+        private Builder() {}
+
+        public Builder(int titleId, int msgBody, int buttonTextId) {
+            this();
+            this.titleId = titleId;
+            this.messageId = msgBody;
+            this.buttonTextId = buttonTextId;
+        }
+
+
+        public int getTitleId() { return titleId; }
+        public String getTitle() { return title; }
```

Comment:
Exactly. If the ```build()``` method takes a context, then it can do the string conversion, and then we only need getters that return strings. That'll simplify a lot of the code that's in the fragment right now by moving the complexity to the builder.